File: interpreter-for-cpp/src/Runtime/Module.cpp
#include "Module.h"
#include "Executer.h"

using namespace peak::interpreter;

Module::Module(const std::string& name, std::shared_ptr<Executer> executer)
	: _name(name), _executer(executer) {
	_space = _executer->GetSpace();
}
Module::Module(const std::string& name, std::shared_ptr<Space> space)
	: _name(name), _space(space) {
}

bool Module::Execute() {
	if (_executer) {
		return _executer->Execute();
	}
	return true;
}

const std::string& Module::GetName() const {
	return _name;
}

std::shared_ptr<Space> Module::GetSpace() const {
	return _space;
}

File: interpreter-for-cs/src/Runtime/Module.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {

	public class Module {
		private string _name = "";
		private Executer _executer = null;
		private Space _space = null;

		public string name { get => _name; }
		public Space space { get => _space; }

		public Module(string name, Executer executer) {
			_name = name;
			_executer = executer;
			_space = _executer.space;
		}
		public Module(string name, Space space) {
			_name = name;
			_space = space;
		}

		public bool Execute() {
			if (_executer != null) {
				return _executer.Execute();
			}
			return true;
		}
	}

} // namespace peak.interpreter